### PR TITLE
ui: delete pki key functionality 

### PR DESCRIPTION
**Pull request by hellobontempo**
delete-key

### CODE REVIEW

File: ui/app/models/pki/key.js [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
   @attr('string') keyName;
   @attr('string') keyType;
-  @attr('string', { detailsLabel: 'Key bit length' }) keyBits; // TODO confirm with crypto team to remove this field from details page
+  @attr('string') keyBits;
 
-  // TODO refactor when field-to-attrs util is refactored as decorator
-  constructor() {
-    super(...arguments);
-    this.formFields = expandAttributeMeta(this, ['keyId', 'keyName', 'keyType', 'keyBits']);
+  get backend() {
```
Comment (hellobontempo) [inline ui/app/models/pki/key.js:16]: Is it bad for this to be a getter? In other situations we've added it as an attribute in the adapter - but I felt it sometimes got lost there. 

Happy to move it there instead!

File: ui/lib/pki/addon/components/pki-key-details.ts [hunk 2]
```diff
@@ -1,27 +0,0 @@
-import { action } from '@ember/object';
```
Comment (hellobontempo) [inline ui/lib/pki/addon/components/pki-key-details.ts:1]: not deleted, just moved into the `page/` folder

File: ui/app/models/pki/key.js [hunk 3]
```diff
@@ -1,18 +1,19 @@
 import Model, { attr } from '@ember-data/model';
-import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+import { inject as service } from '@ember/service';
+import { withFormFields } from 'vault/decorators/model-form-fields';
 
+@withFormFields(['keyId', 'keyName', 'keyType', 'keyBits'])
```
Comment (hashishaw) [inline ui/app/models/pki/key.js:5]: ✨ 🤩

File: ui/app/models/pki/key.js (hunk 1 above)
Comment (hashishaw) [reply ui/app/models/pki/key.js:16]: Ooh I love this! I wonder if we can standardize everywhere else 👀

File: ui/lib/pki/addon/components/page/pki-key-details.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    rollbackAttributes: () => void;
+    destroyRecord: () => void;
+    backend: string;
+    keyName: string;
+    keyId: string;
+  };
+}
+
+export default class PkiKeyDetails extends Component<Args> {
+  @service declare router: { transitionTo: (route: string) => void };
```
Comment (hashishaw) [inline ui/lib/pki/addon/components/page/pki-key-details.ts:16]: Would love to see the types here moved to either a typescript declare module, or imported directly from some stub types folder. Cloud-UI also has a pattern of doing `@service declare readonly myService` but I'm not finding any luck with why. Couldn't hurt though!

File: ui/tests/integration/components/pki/page-key-details-test.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    this.owner.lookup('service:flash-messages').registerTypes(['success', 'danger']);
+    this.store = this.owner.lookup('service:store');
+    this.model = this.store.createRecord('pki/key', {
+      keyId: '724862ff-6438-bad0-b598-77a6c7f4e934',
+      keyType: 'ec',
+      keyName: 'exported-key-2',
+    });
+  });
+
+  test('it renders the page component', async function (assert) {
```
Comment (hashishaw) [inline ui/tests/integration/components/pki/page-key-details-test.js:22]: Should we add a test that confirms the delete action works?

**Review by hashishaw — COMMENTED** [review]
Great work! A couple small items to follow up on. Love to see it! 👏

File: ui/lib/pki/addon/components/page/pki-key-details.ts (hunk 4 above)
Comment (hellobontempo) [reply ui/lib/pki/addon/components/page/pki-key-details.ts:16]: Okay! I wasn't sure how to do the module way. I had noticed the other way for this was: 
```js
 import RouterService from '@ember/routing/router-service';
 service declare readonly router: RouterService;
```

what do you think?

**Review by hashishaw — APPROVED** [review]
Beautiful! Thank you! 🙌

**Pull request merged**